### PR TITLE
Document LFRic mesh properties that kernels can request (towards 18)

### PR DESCRIPTION
**Pull request by arporter**

### CODE REVIEW

**Comment by arporter** [comment]
This PR is purely for the documentation - to make sure I've got things right before I proceed to implement...

**Comment by arporter** [comment]
Suitable for either Tom or Iva :-)

**Comment by TeranIvy** [comment]
Happy to take it!

File: doc/user_guide/dynamo0p3.rst [hunk 1]
```diff
@@ -1184,6 +1179,35 @@ outward_normals_to_faces             Array of outward-pointing normals for each
...
+
+A kernel that requires properties of the LFRic mesh object specifies
+those properties through the ``meta_mesh`` metadata entry. (If no
+mesh properties are required then this metadata should be omitted.)
+Consider the following example kernel metadata::
+
+  type, extends(kernel_type) :: testkern_type
+    type(arg_type), dimension(2) :: meta_args = &
+        (/ arg_type(gh_field, gh_read, w1),     &
+	   arg_type(gh_field, gh_write, w0) /)
```
[the remark this answers is not in the document]
Comment (arporter) [reply doc/user_guide/dynamo0p3.rst:1193]: Done.

File: doc/user_guide/dynamo0p3.rst [hunk 2]
```diff
@@ -1184,6 +1179,35 @@ outward_normals_to_faces             Array of outward-pointing normals for each
...
+  end type testkern_type
+
+This metadata specifies that the ``testkern_type`` kernel requires one
+property of the mesh. There is currently one supported property:
+
+======================= ==================================================
+Name                    Description
+======================= ==================================================
+adjacent_face           Local ID of a neighbouring face in each adjacent
+                        cell indexed as (face).
```
[the remark this answers is not in the document]
Comment (arporter) [reply doc/user_guide/dynamo0p3.rst:1208]: Done.

File: doc/user_guide/dynamo0p3.rst [hunk 3]
```diff
@@ -1408,7 +1432,16 @@ rules, along with PSyclone's naming conventions, are:
       a rank-2 integer array of type ``i_def`` with dimensions
       ``(3, nfaces_re)``.
 
-6) If Quadrature is required (``gh_shape = gh_quadrature_*``) then, for
+6) If the ``adjacent_face`` mesh property is required then:
+
+   1) If the number of horizontal cell faces obtained from the reference
+      element (``nfaces_re_h``) is not already being passed to the kernel (due
+      to rule 5 above) then supply it here. This is an integer of kind
```
[the remark this answers is not in the document]
Comment (arporter) [reply doc/user_guide/dynamo0p3.rst:1439]: Nope, no "the" required :-)

File: doc/user_guide/dynamo0p3.rst [hunk 4]
```diff
@@ -1408,7 +1432,16 @@ rules, along with PSyclone's naming conventions, are:
...
+      ``i_def``.
```
[the remark this answers is not in the document]
Comment (arporter) [reply doc/user_guide/dynamo0p3.rst:1440]: Done. I've left the definition here for clarity.

**Comment by arporter** [comment]
This PR is purely for the documentation. Once it's on I will do a separate PR for the implementation.

**Comment by arporter** [comment]
Ready for another look now.

**Pull request merged**